### DISCUSSION TITLE
Add unikernel support to restrict attack surface

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#5404] [Q&A] [answered]
https://www.phoronix.com/news/Linux-Unikernel-RFC

https://github.com/unikernelLinux/ukl

Maybe in future we can start protect userspace apps inside unikernel.

More optimized and secure.

Thanks and

Best regards

**Comment by kmk3** [comment]
Hello, this seems more like a discussion topic than an actionable issue, so I'm
converting it into a discussion.

**Comment by topimiettinen** [comment, marked as answer]
Unikernel looks like a performance feature and not security oriented at all. I can't see how running programs as superuser and weakening user/kernel boundary would improve security. A bug in the program would lead to breach into kernel mode more easily.
